src/components/tasks.js

```diff
@@ -0,0 +1,12 @@
+import React, { useState } from "react";
+import { Outlet, useLocation, Navigate} from "react-router-dom";
+
+const Tasks = () => {
+    const location = useLocation();
+    
+    return (
+        location.state !== null ? <div>Task Page<Outlet/></div> : <Navigate to="/sign_in"/>
```

Comment:
We can do without the entire boolean check, `location.state !== null`. Instead we can just do a `location?.state?.user && ...`. The `?.` will evaluate to null if the next field does not exist.

Reply:
Actually, can we we move the navigate out of this return? This component is going to get really complicated in the next couple of weeks. We can do something like this, 

```javascript
const { state } = useLocation();

    if (!state || !state?.user) {
        return <Navigate to="/sign-in" />;
    }

    return (
        <div>
            Task Page
            <Outlet />
        </div>
    );
```